packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+// @ts-check
+
+import { AmountMath, AssetKind } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { fit, M, makeCopyBag, makeStore } from '@agoric/store';
+import { assertProposalShape } from '@agoric/zoe/src/contractSupport/index.js';
+
+const { details: X } = assert;
+
+export const KW = harden({
+  /** seat keyword for use in offers to return an attestation. */
+  Attestation: 'Attestation',
+});
+
+/**
+ * Find-or-create value in store.
+ *
+ * @type {<K, V>(store: Store<K, V>, key: K, thunk: () => V) => V}
+ */
+const getOrElse = (store, key, make) => {
+  if (store.has(key)) {
+    return store.get(key);
+  }
+  const value = make();
+  store.init(key, value);
+  return value;
+};
+
+/**
+ * To support `makeAttestation`, we directly mint attestation payments.
+ * We still use a ZCFMint because returning attestations is done
+ * through Zoe offers.
+ *
+ * @param {ZCF} zcf
+ * @param {ZCFMint} zcfMint
+ * @param {Amount} amountToMint
+ * @returns {Promise<Payment>}
+ */
+const mintZCFMintPayment = (zcf, zcfMint, amountToMint) => {
+  const { userSeat, zcfSeat } = zcf.makeEmptySeatKit();
+  zcfMint.mintGains(harden({ [KW.Attestation]: amountToMint }), zcfSeat);
+  zcfSeat.exit();
+  return E(userSeat).getPayout(KW.Attestation);
+};
+
+/**
+ * Make an ZCFMint attentuated so that minting represents
```

Comment:
```suggestion
 * Make an attentuated ZCFMint so that minting represents
```

Reply:
Hm. The "so that ..." goes with "attenuated". I think this change makes it less clear.

Reply:
Maybe I misunderstood the intent of the statement. Is the goal to provide a subset of ZCFMint powers? 

Related, an `@returns` may read better. (Though we might need to disable the eslint rule that requires every JSDoc param tag to have a type, so that it can be inferred by what's actually returned.)